### DISCUSSION TITLE
Static Code Analysis Results

### DISCUSSION THREAD

**Discussion by mm-mbinder** [nlohmann/json#4075] [General]
Hello,

I've included this library in one of our projects, which has a SonarQube analysis active. SonarQube reported some bugs and security hotspots in the included header file. Since i don't have the capabilities or knowledge to evaluate or fix any of the issues, i thought it would be at least a good idea to report the issues here.
For context the single header release from version 3.11.2 was included/analyzed here. MSVC143 is used as compiler.

Bugs:
- [Line 6164](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L6164) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-1048/)
![image](https://github.com/nlohmann/json/assets/74175538/af2fac9a-97fc-4e49-a164-b09d3b53abd1)

- [Line 14715](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L14715) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-6458/)
![image](https://github.com/nlohmann/json/assets/74175538/1fc9cef5-0201-46ef-ad1a-a797024fa626)

- [Line 18943](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L18943) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-5417/)
![image](https://github.com/nlohmann/json/assets/74175538/312bed15-e631-41c3-869d-3b89ad790850)

- [Line 18958](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L18958) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-5417/)
![image](https://github.com/nlohmann/json/assets/74175538/20dffa09-bd1c-406d-a915-cfa861630b21)

- [Line 20003](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L20003) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-6458/)
![image](https://github.com/nlohmann/json/assets/74175538/ae555c8d-a80e-4a8f-9f00-8a14e7cc0d5a)

- [Line 20421](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L20421) and [Line 19742](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L19742) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-1048/)
![image](https://github.com/nlohmann/json/assets/74175538/26702f56-2d6b-43e5-b695-cf93564a15ca)
![image](https://github.com/nlohmann/json/assets/74175538/3dfcf556-ba13-45cd-9ae0-da901fd6c3ba)

Security Hotspots:
- [Line 4183](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L4183) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-5813/)
![image](https://github.com/nlohmann/json/assets/74175538/bf01bd86-4d28-47df-ae52-0a7b84326409)

- [Line 6520](https://github.com/nlohmann/json/blob/master/single_include/nlohmann/json.hpp#L6520) - [SonarQube Description](https://rules.sonarsource.com/cpp/RSPEC-5813/)
![image](https://github.com/nlohmann/json/assets/74175538/d1527183-f0ed-48ba-a81f-115d4c9adbc8)

**Comment by nlohmann** [comment]
Thanks for reporting this — the SonarQube finding descriptions/links didn't come through in the post, and the line numbers reference the 3.11.2 single-include header. Could you re-share the actual rule IDs/messages (not just line numbers) for the bugs and security hotspots? Without the specific messages it's hard to tell whether these are real issues or SonarQube false positives on header-only template-heavy code (which is common).

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
